varcode/variant.py

```diff
@@ -248,18 +251,18 @@ def __eq__(self, other):
             self.ensembl == other.ensembl)
 
     def __getstate__(self):
-        fields = self.fields()
-        return {
-            field: fields[i] for (i, field) in enumerate(fields._fields)
-        }
+        # When pickle.load'ing, we want the original values vs. the normalized values.
+        # Normalization will happen upon object creation.
+        return [self.contig, self.original_start, self.original_ref, self.original_alt, self.ensembl.release]
 
     def __setstate__(self, fields):
-        # This field require special logic.
-        self.ensembl = cached_release(fields.pop("release"))
+        # Special logic needed for JSON serialization, since EnsemblRelease
+        # is not JSON serializable.
+        # TODO: this does not work for Genome objects; only EnsemblRelease.
+        ensembl = cached_release(fields.pop())
```

Comment:
Sure, go for it.

On Mon, Apr 25, 2016 at 4:51 PM, Tavi Nathanson <notifications@github.com>
wrote:

> In varcode/variant.py
> <https://github.com/hammerlab/varcode/pull/146#discussion_r60986731>:
>
> >
> >      def __setstate__(self, fields):
> > -        # This field require special logic.
> > -        self.ensembl = cached_release(fields.pop("release"))
> > +        # Special logic needed for JSON serialization, since EnsemblRelease
> > +        # is not JSON serializable.
> > +        # TODO: this does not work for Genome objects; only EnsemblRelease.
> > +        ensembl = cached_release(fields.pop())
>
> Agreed, that seems like a problem; mind if I create a separate issue since
> it's an old bug that isn't introduced here?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/hammerlab/varcode/pull/146/files/6a6d298d48f473cb30267b1040ce436006b3d43a#r60986731>
>